pkg/api/v1/router_server_secrets.go

```diff
@@ -0,0 +1,149 @@
+package serverservice
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"go.hollow.sh/serverservice/internal/dbtools"
+	"go.hollow.sh/serverservice/internal/models"
+)
+
+func (r *Router) serverSecretGet(c *gin.Context) {
+	mods := []qm.QueryMod{
+		models.ServerSecretWhere.ServerID.EQ(c.Param("uuid")),
+		qm.InnerJoin(fmt.Sprintf("%s as t on t.%s = %s.%s",
+			models.TableNames.ServerSecretTypes,
+			models.ServerSecretTypeColumns.ID,
+			models.TableNames.ServerSecrets,
+			models.ServerSecretColumns.ServerSecretTypeID,
+		)),
+		qm.Where(fmt.Sprintf("t.%s=?", models.ServerSecretTypeColumns.Slug), c.Param("slug")),
+		qm.Load(models.ServerSecretRels.ServerSecretType),
+	}
+
+	dbS, err := models.ServerSecrets(mods...).One(c.Request.Context(), r.DB)
+	if err != nil {
+		dbErrorResponse(c, err)
+		return
+	}
+
+	decryptedValue, err := dbtools.Decrypt(c.Request.Context(), r.SecretsKeeper, dbS.Value)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, &ServerResponse{Message: "error decrypting secret value", Error: err.Error()})
+		return
+	}
+
+	secret := &ServerSecret{
+		SecretType: dbS.R.ServerSecretType.Slug,
+		Value:      decryptedValue,
+		CreatedAt:  dbS.CreatedAt,
+		UpdatedAt:  dbS.UpdatedAt,
+	}
+
+	secret.ServerID, err = uuid.Parse(dbS.ServerID)
```

Comment:
should we parse this as the first thing that happens in the function? that way we sanitize input AND would already detect an error early in the function's lifetime.

Reply:
This can't actually fail. The database stores the ID as a UUID, but it gets brought in as a string, so we have to convert it to a UUID. I could use MustParse but that would throw a fatal error that we wouldn't catch. So we just parse and handle the error, but if that database returns a bad UUID when it is stored as a UUID type we likely have a pretty big problem.

Reply:
gotcha, I still think we should move this to the top. Let's take a defensive approach and take "solar flares" into account. So, we could build the `secret.ServerSecret` object in the beginning of the function. Add the `ServerID` by parsing the given UUID, and then proceed. If somehow (magically, solar flares, CPU bit flip, etc.) there would be an error parsing, we would catch it immediately.